### PR TITLE
Karel/clippy forbid panics

### PR DESCRIPTION
**Pull request by KaiserKarel**

### CODE REVIEW

**Comment by KaiserKarel** [comment]
This should trigger loads of failures right now, mainly related to slice indexing as well as usage of some other stuff.

**Review by seunlanlege — APPROVED** [review]

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -49,8 +49,18 @@
...
 
-#![cfg_attr(not(test), warn(clippy::disallowed_method, clippy::indexing_slicing))] // allow in tests
-#![warn(clippy::unseparated_literal_suffix, clippy::disallowed_type)]
+#![cfg_attr(
+	not(test),
+	warn(
+		clippy::disallowed_method,
+		clippy::disallowed_type,
+		clippy::indexing_slicing,
+		clippy::todo,
```
Comment (dzmitry-lahoda) [inline frame/assets/src/lib.rs:58]: how we should mark to do items? todo was nice way to do it

Comment (seunlanlege) [reply frame/assets/src/lib.rs:58]: I would say that we shouldn't mark todo items, and prs should either finish the implementation or not be merged

Comment (KaiserKarel) [reply frame/assets/src/lib.rs:58]: I can remove the todo macro from the crates, and enable it in CI for releases if that makes more sense. But I think for PR's to main, `todo` should be handled by then to be honest.

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Pull request merged**